noir-projects/noir-contracts/contracts/ecdsa_public_key_note/src/lib.nr

```diff
@@ -80,18 +80,10 @@ impl NoteInterface<ECDSA_PUBLIC_KEY_NOTE_LEN> for EcdsaPublicKeyNote {
     }
 
     fn compute_note_hash(self) -> Field {
-        let packed_content = self.pack_content();
-        std::embedded_curve_ops::multi_scalar_mul(
-            [Gx_1, Gx_2, Gy_1, Gy_2, G_owner, G_slot],
-            [
-            from_field_unsafe(packed_content[0]),
-            from_field_unsafe(packed_content[1]),
-            from_field_unsafe(packed_content[2]),
-            from_field_unsafe(packed_content[3]),
-            from_field_unsafe(packed_content[4]),
-            from_field_unsafe(self.get_header().storage_slot)
-        ]
-        ).x
+        // We use Poseidon2 instead of multi-scalar multiplication (MSM) here since this is not a partial note
+        // and therefore does not require MSM's additive homomorphism property. Additionally, Poseidon2 uses fewer
+        // constraints.
+        poseidon2_hash_with_separator(self.pack_content(), NOTE_HASH_SEPARATOR)
```

Comment:
How come you have control over this function? Is it because the note was defined with `#[note_custom_interface]`? Does that annotation somehow prevent the note interface macro creation function from running, and force you to write your own? Pretty cool.

Reply:
Exactly as you say. #[note_custom_interface] makes macros not generate the NoteInterface functions.